Projects/Server.Tests/Tests/Network/RingPipeTests.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using Server.Network;
+using Xunit;
+
+namespace Server.Tests.Network;
+
+[Collection("Sequential Tests")]
+public class RingPipeTests
+{
+    [Fact]
+    public void TestWriting()
+    {
+        var pageSize = (uint)Environment.SystemPageSize;
+        var pipe = new RingPipe(pageSize);
+
+        var span = pipe.Writer.TryGetMemory();
+
+        for (var i = 0; i < span.Length; i++)
+        {
+            span[i] = (byte)(i % 256);
+        }
+
+        pipe.Writer.Advance(pageSize - 10);
```

Comment:
How come this is -10?

Reply:
I am testing a cross-over point, so I picked this arbitrarily. I just updated the test, so check that.